subprojects/core/src/main/java/org/gradle/execution/taskgraph/DefaultTaskExecutionPlan.java

```diff
@@ -538,10 +539,16 @@ public boolean executeWithTask(WorkerLease parentWorkerLease, final Action<TaskI
     }
 
     private ResourceLock getProjectLock(TaskInfo taskInfo) {
+        if (projectLocks.containsKey(taskInfo)) {
```

Comment:
:o: Can this method be accessed concurrently? If so it should probably be synchronized, or use a concurrent hashmap.